freqtrade/freqai/freqai_interface.py

```diff
@@ -991,3 +968,26 @@ def predict(
         :do_predict: np.array of 1s and 0s to indicate places where freqai needed to remove
         data (NaNs) or felt uncertain about data (i.e. SVM and/or DI index)
         """
+
+    # deprecated functions
+    def data_cleaning_train(self, dk: FreqaiDataKitchen, pair: str):
+        """
+        throw deprecation warning if this function is called
+        """
+        ft = "https://www.freqtrade.io/en/latest"
```

Comment:
This is used twice - and defined twice - so it should be extracted by at least one level (to only define this once).
I also think it should point to stable.
While it'll become an invalid link until the next release - i think users should for the most part be using "stable" - not "latest" from a docs perspective.
Latest can contain features not yet available with the release version - which in my experience is what most people are using.

Reply:
Done, moved to `constants.py`